### DISCUSSION TITLE
Signature validation failed

### DISCUSSION THREAD

**Discussion by morohon** [strongswan/strongswan#1247] [Q&A]
Hi all.
I'm trying to set up ikev2 vpn on my vps server with certificate authorization. I generate certificates through easyrsa (I took the latest version and generated CA, server cert and key, client cert and key). When I try to connect from an android device, I get an error in the log: "signature validation failed, looking for another key" and I can't figure out what's wrong.
At the moment, the goal is to raise vpn, which will be the gateway of all traffic.
Below I attach all the configurations that are used.
I would be grateful for help.

Strongswan version: 
```
Linux strongSwan U5.9.6/K4.18.0-305.7.1.el8_4.x86_64
University of Applied Sciences Rapperswil, Switzerland
See 'strongswan --copyright' for copyright information.
```

OS: Oracle Linux Server release 8.6

Settings on the client:
Server address: vpn.example.com
IPSec ID: vpn.example.com
User certificate: me-phone.vpn.example.com
CA certificate: ca.vpn.example.com
Server certificate: received from the server

swanctl -L here: https://gist.github.com/morohon/2f4ecfcd099d2e1af7a5d431cc0ce622
swanctl --list here: https://gist.github.com/morohon/d94f46d156bc7ab7205c7b517242ca22
swanctl.conf here: https://gist.github.com/morohon/4133119d6f5e007d11ea82bb26eb4f35
swanctl logs here: https://gist.github.com/morohon/1b161e80c68848ef6d7f39c9dfcc9fb8

**Comment by Thermi** [comment]
Can you check if the client certificates verifies correctly against your CA if you use `pki --verify`?

**Reply by morohon** [reply]
Yes:
https://gist.github.com/morohon/50788abc9eab53c9161d4794c38e979e

**Reply by Thermi** [reply]
OK. Because it'S Android, I presume there's some problem with its code again. Please try different algorithms for the key and different digest types.

**Comment by tobiasbrunner** [comment]
You configured the wrong identity on the client. It authenticates as `vpn.example.com`:

    16[CFG] looking for peer configs matching y.y.y.y[vpn.example.com]...x.x.x.x[vpn.example.com]

instead of (presumably) `me-phone.vpn.example.com`. This causes the server to select its own certificate, which obviously can't verify the client's AUTH payload. Make sure you configure the correct identity on the client.
